Analysis/Tasks/run2Matcher.cxx

```diff
@@ -22,14 +22,8 @@ using namespace o2::framework::expressions;
 struct Run2Matcher {
   BuildsExclusive<aod::Run2MatchedExclusive> matched_e;
   Builds<aod::Run2MatchedSparse> matched_s;
-  void init(o2::framework::InitContext&)
-  {
-  }
-};
-
-struct BCMatcher {
-  BuildsExclusive<aod::BCCollisionsExclusive> matched_e;
-  Builds<aod::BCCollisionsSparse> matched;
+  BuildsExclusive<aod::BCCollisionsExclusive> bc_e;
```

Comment:
Is that a good idea to build all indices in one task? There is no optimization for building multiple index tables simultaneously and if only one of the tables is requested, there is no point in building all of them.

Reply:
Then we have to put them in different workflows even. Before it was two devices in one workflow.

Reply:
Could we do the index building similarly to the expression columns automatically and once such a table is subscribed?

Reply:
Yes, it is possible to use the same trick as is done with data model extended tables, though it requires giving these tables a separate origin and hardcoding their names.

Reply:
Very good. Yes, we can have an origin for automatic spawning. What needs to be hardcoded in the table names?

Reply:
Similar to what we have in `AODReaderHelpers.cxx` lines 78-86

Reply:
OK, so each case. If users add custom ones, they would need to produce them themselves. I think this is totally OK.

Reply:
Let's merge this first and I will add `internal-dpl-index-builder` device later